### PR TITLE
Login: Allow custom name and icon for social providers

### PR DESCRIPTION
**Pull request by jkroepke**
**What is this feature?**

This PR allows the modification of social login providers

<details>
<summary>Details</summary>

image

</details>

**Why do we need this feature?**

Users may known to `Login with company IDP` or `Login with Azure AD` instead `Login with Microsoft`

**Who is this feature for?**

End Users

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer**:

Okta social provider already supports this. The field name and icon is already availible for all social providers, but the frontend component does not care about it.

### CODE REVIEW

**Review by kalleep — COMMENTED** [review]
Thanks for the pr. Could you also update the sample.ini file with these values?

File: conf/defaults.ini [hunk 1]
```diff
@@ -589,8 +597,9 @@ skip_org_role_sync = false
 
 #################################### Azure AD OAuth #######################
 [auth.azuread]
-name = Azure AD
-enabled = false
+name = Microsoft
+icon = microsoft
+enabled = true
```
[the remark this answers is not in the document]
Comment (jkroepke) [reply conf/defaults.ini]: Left-over from local testing! Thanks!

**Comment by kalleep** [comment]
Sadly we cannot accept the pr, some providers has guidelines on how these buttons should look and feel.
Some references:
https://learn.microsoft.com/en-us/azure/active-directory/develop/howto-add-branding-in-azure-ad-apps
https://developers.google.com/identity/branding-guidelines

**Comment by jkroepke** [comment]
@kalleep I'm bit sad to hear this.

The restrictions you mentioned are for marketplace related instances, but not for private instances. After merging the PR, the Sign button are still compliant for the mention guidelines. From grafana point of view, the Administrator of Grafana is now responsible for been complaint. 

Please, re-think about this. We have a Single Tenant Application on Azure AD and 2 Tenants to connect.

I'm able to change the label of the generic oauth provider while I'm not able to change it for the azuread provider.

Right now, we on our login screen

- Sign in with Tenant A
- Sign in with Microsoft

Which confuses or end users.
---

I'm asking my self, why other solutions are less strict here and allows end-users to configure the label of the button, for example [Gitlab](https://docs.gitlab.com/ee/integration/azure.html), [ArgoCD](https://argo-cd.readthedocs.io/en/stable/operator-manual/user-management/microsoft/#log-in-to-argocd-ui-using-sso).

Please @kalleep, re-think about this, if Grafana should enforce this on application level OR if its okay the move guideline compliance to grafana end-users like me.

**Comment by danielkenlee** [comment]
Hi @jkroepke, we discussed internally and have decided to allow merging the PR. You made some reasonable arguments implementing this change. Grafana also doesn't want to be in the position of enforcing the guidelines from the identity providers and rightly the responsibility should sit with end users / admins. We strive to provide consistent and powerful tooling to our users and recognise value of this of configuration option.

**Comment by kalleep** [comment]
@jkroepke before we can approve the pr, could you please also update the sample.ini file with the new options?

**Comment by jkroepke** [comment]
@kalleep done 👍

**Review by linoman — APPROVED** [review]
Great job @jkroepke

**Review by kalleep — APPROVED** [review]

**Pull request merged**